File: solutions/29.py
import time

def count_unique_powers (max):
  arr = []
  for i in range(2, max + 1):
    for j in range(2, max + 1):
      current = i ** j
      if current not in arr:
        arr.append(current)
  return len(arr)

start = time.time()
print(count_unique_powers(100))
end = time.time()
print(end - start)
# runtime: 0.442s

File: solutions/29.js
let count = 0
let arr = []
let current
let found = false
for (let i = 2; i < 6; i++) {
  for (let j = 2; j < 6; j++) {
    found = false
    current = Math.pow(i, j)
    for (let k = 0; k < arr.length; k++) {
      if (arr[k] === current) {
        found = true
        break
      }
    }
    if (!found) {
      count++
      arr.push(current)
    }
  }
}
console.log(count)